tests/functions/test_encode_categorical.py

```diff
@@ -98,179 +99,139 @@ def test_check_presence_column_names_in_kwargs(df):
     do not exist in the dataframe.
```

Comment:
I don't think this test (`test_check_presence_column_names_in_kwargs`) requires `hypothesis`, since it is only checking validity of column names.
Let's remove the hypothesis line `@given(df=df_strategy())` to save some runtime on our tests.

And similarly for the test above this one (`test_both_column_names_kwargs`) as well.